### PR TITLE
Add model DTOs

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

File: dto/src/main/java/org/jboss/pnc/dto/model/Build.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    private final ProjectRef project;
+
+    private final RepositoryConfigurationRef repository;
+
+    private final BuildEnvironmentRef buildEnvironmentId;
+
+    private final Map<String, String> attributes;
+
+    private final GroupBuildRef groupBuild;
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/Build.java]: It was in the original BuildRecordRest, but I don't mind removing it.

File: dto/src/main/java/org/jboss/pnc/dto/model/Build.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * The IDs of the build record sets which represent the builds performed for a milestone to which this build record belongs
+     */
+    private final ProductMilestoneRef productMilestone;
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/Build.java]: It was in the original BuildRecordRest, but I don't mind removing it.

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    @RefHasId(groups = {WhenCreatingNew.class, WhenUpdating.class})
+    protected final BuildEnvironmentRef environment;
+
+    private final Set<Integer> dependencyIds;
+
+    @RefHasId(groups = {WhenCreatingNew.class, WhenUpdating.class}, optional = true)
+    private final ProductVersionRef productVersion;
+
+    private final Set<Integer> groupBuildIds;
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java]: Yes, this should be `groupConfigIds`

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.Value;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@Value
+public class BuildConfiguration extends BuildConfigurationRef {
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java]: It's on the constructor

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import lombok.Value;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@Value
+public class BuildConfigurationRevision extends BuildConfigurationRevisionRef {
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java]: It's on the constructor

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    private final RepositoryConfigurationRef repositoryConfiguration;
+
+    private final ProjectRef project;
+
+    private final BuildEnvironment environment;
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java]: BuildEnvironmentRef doesn't exist.  (if it does, BuildEnvironment would be empty)

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/BuildConfigurationRevision.java]: Dependencies are defined on the BC level, they are not tracked. The same applies for productVersion and groupConfis AFAIK.

File: dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@Value
+public class GroupConfig extends GroupConfigRef {
+
+    @RefHasId(groups = {WhenCreatingNew.class, WhenUpdating.class}, optional = true)
+    private final ProductVersionRef productVersion;
+
+    private final List<Integer> buildConfigurationIds;
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java]: What? It is a list.

File: dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import lombok.Value;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@Value
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java]: It's on the constructor.

File: dto/src/main/java/org/jboss/pnc/dto/model/GroupBuild.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import lombok.Value;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@Value
+public class GroupBuild extends GroupBuildRef {
```
[the remark this answers is not in the document]
Comment (janinko) [reply dto/src/main/java/org/jboss/pnc/dto/model/GroupBuild.java]: It's on the constructor.

File: dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (jbartece) [reply dto/src/main/java/org/jboss/pnc/dto/model/GroupConfig.java]: The editor removed the essential part of the comment. List BuildConfigurationRef

**Comment by matejonnet** [comment]
What is the meaning of the model DTO ?

File: dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+    @RefHasId(groups = {WhenCreatingNew.class, WhenUpdating.class})
+    protected final BuildEnvironment environment;
+
+    private final Set<Integer> dependencyIds;
+
+    @RefHasId(groups = {WhenCreatingNew.class, WhenUpdating.class}, optional = true)
+    private final ProductVersionRef productVersion;
+
+    private final Set<Integer> groupConfigIds;
```
Comment (jbartece) [inline dto/src/main/java/org/jboss/pnc/dto/model/BuildConfiguration.java]: Why not to use GroupConfigRefs?

**Comment by janinko** [comment]
@matejonnet Model are the classes that represents PNC data model

**Comment by janinko** [comment]
@matejonnet Model are the classes that represents PNC data model

**Comment by janinko** [comment]
@matejonnet Model are the classes that represents PNC data model

**Pull request merged**